services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,302 @@
+
+		// create storageClass
+		createCephBlockPoolStorageClass(w, client, ExpandStorage.StorageClassDetails.Name, ExpandStorage.PoolDetails.PoolName, ExpandStorage.StorageClassDetails.ReclaimPolicy, ExpandStorage.StorageClassDetails.VolumeBindingMode, ExpandStorage.StorageClassDetails.EnableStorageClassEncryption, ExpandStorage.StorageClassDetails.EncryptionKMSID, namespace)
+	} else {
+		// create cephFilesystem
+		createCephFilesystem(w, client, ExpandStorage.PoolDetails.PoolName, ExpandStorage.LsoStorageClassName, ExpandStorage.PoolDetails.DataProtectionPolicy, ExpandStorage.PoolDetails.EnableCompression, ExpandStorage.PoolDetails.FailureDomain, storageCluster)
+
+		// create storageClass
+		createCephFilesystemStorageClass(w, client, ExpandStorage.StorageClassDetails.Name, ExpandStorage.PoolDetails.PoolName, ExpandStorage.StorageClassDetails.ReclaimPolicy, ExpandStorage.StorageClassDetails.VolumeBindingMode, ExpandStorage.PoolDetails.FilesystemName, namespace)
+	}
+
+}
+
+func updateStorageCluster(w http.ResponseWriter, client client.Client, lsoStorageClassName string, enableEncryption bool, storage, count, replica int, storageCluster *ocsv1.StorageCluster) {
+	klog.Infof("Updating storageCluster %q", storageCluster.Name)
+	storageQty := resource.MustParse(fmt.Sprintf("%d", storage))
+	deviceSet := ocsv1.StorageDeviceSet{
+		Name:        lsoStorageClassName,
+		Count:       count,
+		Replica:     replica,
+		Portable:    false,
+		Encrypted:   &enableEncryption,
+		DeviceClass: lsoStorageClassName,
+		DataPVCTemplate: corev1.PersistentVolumeClaim{
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      []corev1.PersistentVolumeAccessMode{"ReadWriteOnce"},
+				StorageClassName: &lsoStorageClassName,
+				Resources: corev1.VolumeResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: storageQty,
+					},
+				},
+			},
+		},
+	}
+
+	mutateFunc := func() error {
+		storageCluster.Spec.StorageDeviceSets = append(storageCluster.Spec.StorageDeviceSets, deviceSet)
+		return nil
+	}
+
+	_, err := ctrl.CreateOrUpdate(context.TODO(), client, storageCluster, mutateFunc)
+	if err != nil {
+		klog.Errorf("failed to update storageCluster: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func createCephBlockPool(w http.ResponseWriter, client client.Client, poolName, lsoStorageClassName string, dataProtectionPolicy int, enableCompression bool, namespace, failureDomain string) {
+	compression := "none"
+	if enableCompression {
+		compression = "aggressive"
+	}
+	cephBlockPool := &cephv1.CephBlockPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      poolName,
+			Namespace: namespace,
+		},
+		Spec: cephv1.NamedBlockPoolSpec{
+			PoolSpec: cephv1.PoolSpec{
+				FailureDomain: failureDomain,
+				DeviceClass:   lsoStorageClassName,
+				Replicated: cephv1.ReplicatedSpec{
+					Size:                   uint(dataProtectionPolicy),
```

Comment:
Is this feature enabled for stretch clusters? The pool spec is a bit different in that case, where we need to set `replicasPerFailureDomain: 2`, and the `subFailureDomain`

Reply:
@travisn by any chance Rook takes care of this automatically ??
Asking because UI never sets this on day-2 BlockPools.
Also, I didn't find any reference of it in OCS operator for day-1 BlockPool (unless I missed, @malayparida2000 can u plz confirm ??)

Reply:
Correct we don't have this enabled for `BlockPools` UI currently, so it is not a common scenario,
We can have this added once we have Blockpool UI using it,

Reply:
On the default cephblockpool that ocs-operator creates the Size & the Replica per failure domain field are set accordingly if it's a stretch cluster. https://github.com/red-hat-storage/ocs-operator/blob/c079ab81fd02767ddccc8c24bd324e6c4df89090/controllers/storagecluster/generate.go#L135-L148.

Reply:
If extra blockpools are being created via UI those would not be reconciled by ocs operator. So adding the options during creation as Travis suggests seems to be a good idea, but not sure why UI didn't have this configured till now

Reply:
Added a arbiter check